packages/flutter/lib/src/rendering/sliver_persistent_header.dart

```diff
@@ -17,6 +21,25 @@ import 'sliver.dart';
 import 'viewport.dart';
 import 'viewport_offset.dart';
 
+// Trims the specified edges of the given `Rect` [original], so that they do not
+// exceed the given values.
```

Comment:
I'm tempted to make this a private extension method to avoid the null check. Is using method extensions not recommended?